openghg/store/_populate.py

```diff
@@ -0,0 +1,209 @@
+        >>> add_noaa_obspack(Path("/home/user/obspack_ch4_1_GLOBALVIEWplus_v2.0_2020-04-24"), "surface-flask")
+        {"processed": {"ch4_esp_surface-flask_2_representative.nc":{"ch4": ...}, ...}}
+
+    TODO: At the moment this will exclude all types which we can't process
+    yet e.g. aircraft, shipboard, aircorenoaa. These should be updated once
+    this functionality has been added.
+    '''
+
+    # Options which we can process at the moment (ObsSurface)
+    project_options = {"surface": ["flask", "insitu", "pfp"]}
+
+    project_names = _create_project_names(project_options)
+
+    # Options we can't process at the moment but may be encountered (ObsMobile, ...).
+    # TODO: "tower-insitu" should be able to run through ObsSurface but
+    # these contain multiple heights per file - not sure we are handling this yet
+    project_options_not_implemented_yet = {"tower": ["insitu"],
+                                           "aircraft": ["pfp", "insitu"],
+                                           "shipboard": ["flask"],
+                                           "aircorenoaa": [""]}
+
+    project_names_not_implemented = _create_project_names(project_options_not_implemented_yet)
+
+    # If a specific project has been specified, extract file matching strings
+    if project:
+        if project in project_names:
+            projects_to_read = [project]
+        elif project in project_options:
+            projects_to_read = [name for name in project_names if project in name]
+        elif project in project_options_not_implemented_yet or project in project_names_not_implemented:
+            raise ValueError(f"Functionality to process {project} data has not been implemented yet.")
+        else:
+            raise ValueError(f"Did not recognise input {project} for project")
+    else:
+        projects_to_read = project_names
+        print(f"Reading subset of data from {','.join(projects_to_read)}")
+
+    if isinstance(data_directory, str):
+        data_directory = Path(data_directory)
+
+    # ObsPack may contain nc or txt files
+    # Try and extract one and then the other if possible
+    files = _find_noaa_files(data_directory, ".nc")
+    if len(files) == 0:
+        files = _find_noaa_files(data_directory, ".txt")
+
+    # Find relevant details for each file and call parse_noaa() function
+    processed_summary: Dict[str, Dict] = {}
+    for filepath in files:
+        param = _param_from_filename(filepath)
+        site = param["site"]
+        project = param["project"]
+        measurement_type = param["measurement_type"]
+
+        if project in projects_to_read:
+            processed = ObsSurface.read_file(filepath, site=site, measurement_type=measurement_type, network="NOAA", data_type="NOAA")
+        elif project in project_names_not_implemented:
+            print(f"Not processing {filepath.name} - no standardisation for {project} data implemented yet.")
+            processed = {}
```

Comment:
I get an `unbound variable` issue raised here, will project **always** be in one of these two? Should we have an

```
else:
    processed = {}
```

Reply:
Well project was derived from our defined `project_options` and `project_options_not_yet_implemented` variables which *should* cover all the possibilities we can process.

However, there could be a case for a value not matching, so I can add this in.